catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/IntegerRangeValidator.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Set;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+import ddf.catalog.validation.violation.ValidationViolation.Severity;
+
+public class IntegerRangeValidator implements AttributeValidator {
+    private final long min;
+
+    private final long max;
+
+    public IntegerRangeValidator(final long min, final long max) {
+        this.min = min;
+        this.max = max;
+    }
+
+    @Override
+    public AttributeValidationReport validate(Attribute attribute) {
+        final AttributeValidationReport report = new AttributeValidationReportImpl();
+
+        if (attribute != null) {
+            final String name = attribute.getName();
+            for (final Serializable value : attribute.getValues()) {
+                final boolean valid;
+                if (value instanceof Short) {
```

Comment:
I don't like the way this is currently being handled because we're missing out on other classes like `BigDecimal` and `BigInteger` (also, `DecimalRangeValidator` has been refactored so it works with any `Number` subclass and this class should be able to handle the same).

We can't just do `((Number) value).longValue()` since that could result in incorrect validation for floating-point values.

I'm thinking about refactoring this portion of the method:
```
if (value instanceof Number) {
    double doubleValue = ((Number) value).doubleValue();
    if (doubleValue % 1 == 0) {
        // A double cannot always accurately represent values at the upper range of a long, so we need to get the actual long value.
        long longValue = ((Number) value).longValue();
        valid = checkRange(longValue);
    } else {
        valid = checkRange(doubleValue);
    }
} else {
    continue;
}
```